### PR TITLE
Feat: Edge apis integration

### PR DESCRIPTION
**Pull request by nicktruong**
### Tasks
- [Edge APIs](https://trello.com/c/UjtgYdh7)

### Describe
- Extract `Card` component from `StandardNode`
- `Get` edges on cards and nodes load
- Integrate `Create` and `Update` edge api

### Demo videos
- https://drive.google.com/file/d/1O5gWTNmuu_XpmcDqYsNBzCb7148ewEuv/view?usp=drive_link

### CODE REVIEW

File: src/hooks/apiHooks/queries/useGetEdgeByCardId.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import { axiosClient } from '@/apis';
+import { queryKeys } from '@/constants';
+import type { Edge, DataResponse, ErrorResponse } from '@/interfaces';
+
+export const useGetEdgeByCardOrNodeId = ({
+  id,
+  type,
+}: {
+  id: string;
+  type: 'card' | 'node';
```
Comment (carl-huynh-goldenowl) [inline src/hooks/apiHooks/queries/useGetEdgeByCardId.ts:13]: We can use enum for this

File: src/app/pages/Studio/containers/Canvas/helpers.ts [hunk 2]
```diff
@@ -60,9 +68,27 @@ export const usePrepareHook = () => {
...
-    [setEdges],
+    (params: Connection) => {
+      const options = { ...params, type: 'smoothstep' };
+      const newData = options.source;
+
+      setEdges(eds => {
+        return addEdge(
+          { ...options, data: newData },
+          eds.filter(edge => {
+            console.log(edge.data === newData);
```
Comment (carl-huynh-goldenowl) [inline src/app/pages/Studio/containers/Canvas/helpers.ts:79]: Remove console.log if unnecessary

File: src/app/pages/Studio/components/Card/helpers.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  const edgeData = data?.data;
+
+  const { setEdges } = useContext(CanvasContext);
+
+  useEffect(() => {
+    if (!edgeData || isFetching) return;
+
+    const edge: Edge = {
+      id: edgeData.id,
+      type: 'smoothstep',
```
Comment (carl-huynh-goldenowl) [inline src/app/pages/Studio/components/Card/helpers.ts:22]: Can we use enum for this?

File: src/app/pages/Studio/components/EndNode/helpers.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import { useContext, useEffect } from 'react';
+
+import { useGetEdgeByCardOrNodeId } from '@/hooks';
+
+import { CanvasContext } from '../../contexts';
+import { Edge, addEdge } from 'reactflow';
```
Comment (carl-huynh-goldenowl) [inline src/app/pages/Studio/components/EndNode/helpers.ts:6]: Check import order again

**Review by carl-huynh-goldenowl — CHANGES_REQUESTED** [review]

**Comment by nicktruong** [comment]
Merging this branch after refactor based on comments.

**Pull request merged**